From patchwork Mon Oct  8 15:04:03 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Vitaly Kuznetsov <vkuznets@redhat.com>
X-Patchwork-Id: 10630927
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id D5475112B
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  8 Oct 2018 15:04:09 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id C6376292D4
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  8 Oct 2018 15:04:09 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id BA570295B6; Mon,  8 Oct 2018 15:04:09 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 6DFF6292D4
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  8 Oct 2018 15:04:09 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726458AbeJHWQR (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Mon, 8 Oct 2018 18:16:17 -0400
Received: from mx1.redhat.com ([209.132.183.28]:39628 "EHLO mx1.redhat.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1726078AbeJHWQR (ORCPT <rfc822;kvm@vger.kernel.org>);
        Mon, 8 Oct 2018 18:16:17 -0400
Received: from smtp.corp.redhat.com (int-mx08.intmail.prod.int.phx2.redhat.com
 [10.5.11.23])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mx1.redhat.com (Postfix) with ESMTPS id 081BC40221
        for <kvm@vger.kernel.org>; Mon,  8 Oct 2018 15:04:08 +0000 (UTC)
Received: from vitty.brq.redhat.com (unknown [10.40.205.25])
        by smtp.corp.redhat.com (Postfix) with ESMTP id EB2CF2DE98;
        Mon,  8 Oct 2018 15:04:06 +0000 (UTC)
From: Vitaly Kuznetsov <vkuznets@redhat.com>
To: kvm@vger.kernel.org
Cc: Paolo Bonzini <pbonzini@redhat.com>
Subject: [kvm-unit-tests PATCH] x86: nVMX: fix EPT A/D disabled MMIO access
 test
Date: Mon,  8 Oct 2018 17:04:03 +0200
Message-Id: <20181008150403.8489-1-vkuznets@redhat.com>
X-Scanned-By: MIMEDefang 2.84 on 10.5.11.23
X-Greylist: Sender IP whitelisted,
 not delayed by milter-greylist-4.5.16 (mx1.redhat.com [10.5.110.30]);
 Mon, 08 Oct 2018 15:04:08 +0000 (UTC)
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

When testing MMIO interface to APIC we need to make sure that APIC is in
xAPIC mode (and not x2APIC). KVM started enforcing this since commit
d1766202779e ("x86/kvm/lapic: always disable MMIO interface in x2APIC
mode").

While the 'clean' fix would be to separate L2's APIC from L1's and put
it to xAPIC mode or create a different MMIO mapping just for the test
we lack the required infrastructure in vmx library. As a quick band-aid
put L1's APIC to xAPIC mode for both EPT A/D enabled and disabled tests,
it seems we don't require x2APIC for anything.

Reported-by: kernel test robot <rong.a.chen@intel.com>
Signed-off-by: Vitaly Kuznetsov <vkuznets@redhat.com>
---
 x86/vmx_tests.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/x86/vmx_tests.c b/x86/vmx_tests.c
index b105b23..cc5192f 100644
--- a/x86/vmx_tests.c
+++ b/x86/vmx_tests.c
@@ -1094,6 +1094,8 @@ static int ept_init_common(bool have_ad)
 	install_ept(pml4, (unsigned long)data_page1, (unsigned long)data_page2,
 			EPT_RA | EPT_WA | EPT_EA);
 
+	reset_apic();
+
 	apic_version = apic_read(APIC_LVR);
 
 	ret = pci_find_dev(PCI_VENDOR_ID_REDHAT, PCI_DEVICE_ID_REDHAT_TEST);
